packages/repository/src/__tests__/unit/repositories/relations-helpers/mongo-related-helpers.unit.ts

```diff
@@ -0,0 +1,135 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/repository-tests
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {ObjectID} from 'bson';
+import {belongsTo, Entity, hasMany, model, property} from '../../../..';
+import {
+  isBsonType,
+  deduplicate,
+  normalizeKey,
+  buildLookupMap,
+  reduceAsArray,
+} from '../../../../relations';
+
+describe('unit tests, simulates mongodb env for helpers of ininclusion resolver ', () => {
+  describe('helpers for formating instances', () => {
+    it('isBsonType', () => {
+      const id = new ObjectID();
+      expect(isBsonType(id)).to.be.true();
+    });
+
+    context('deduplicate + isBsonType', () => {
+      it('simple unique array', () => {
+        const id1 = new ObjectID();
+        const id2 = new ObjectID();
+
+        const result = deduplicate([id1, id2]);
+        expect(result).to.deepEqual([id1, id2]);
+      });
+
+      it('multiple items array', () => {
+        const id1 = new ObjectID();
+        const id2 = new ObjectID();
+        const id3 = new ObjectID();
+
+        const result = deduplicate([id3, id1, id1, id3, id2]);
+        expect(result).to.deepEqual([id3, id1, id2]);
+      });
+    });
+  });
+
+  describe('helpers for generating inclusion resolvers', () => {
+    // the tests below simulate mongodb environment.
+    context('normalizeKey + buildLookupMap', () => {
+      it('checks type', async () => {
```

Comment:
I think this title could be more descriptive, e.g. `checks if normalized key is type string` or something else.